### ISSUE TITLE
Unit tests should not rely on networked resources

### ISSUE THREAD

**Issue by gvwilson** [swcarpentry/amy#579]
A couple of our tests appear to rely on network resources (i.e., they hang if there isn't a network connection). We should stub that out somehow so that tests will run offline.

**Comment by pbanaszkiewicz** [comment]
Can you show examples of such tests?

**Comment by gvwilson** [comment]
~~~
# Check that network is turned off.
(py3k)~/s/amy: ping www.google.com
ping: cannot resolve www.google.com: Unknown host

# Run tests (note: all pass when network is on).
(py3k)~/s/amy: make test
python3 manage.py test
Creating test database for alias 'default'...
..........................................................................................................................................E..............
======================================================================
ERROR: test_wrong_url (workshops.test.test_util.TestParsingEventHeaders)
----------------------------------------------------------------------
Traceback (most recent call last):
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connection.py", 
line 135, in _new_conn
     (self.host, self.port), self.timeout, **extra_kw)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/util/connection.py", 
line 66, in create_connection
     for res in socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM):
   File "/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/socket.py", 
line 533, in getaddrinfo
     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", 
line 559, in urlopen
     body=body, headers=headers)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", 
line 345, in _make_request
     self._validate_conn(conn)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", 
line 782, in _validate_conn
     conn.connect()
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connection.py", 
line 215, in connect
     conn = self._new_conn()
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connection.py", 
line 144, in _new_conn
     self, "Failed to establish a new connection: %s" % e)
requests.packages.urllib3.exceptions.NewConnectionError: 
<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 
0x108343f28>: Failed to establish a new connection: [Errno 8] nodename 
nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/adapters.py", 
line 370, in send
     timeout=timeout
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", 
line 609, in urlopen
     _stacktrace=sys.exc_info()[2])
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/packages/urllib3/util/retry.py", 
line 271, in increment
     raise MaxRetryError(_pool, url, error or ResponseError(cause))
requests.packages.urllib3.exceptions.MaxRetryError: 
HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max 
retries exceeded with url: /test/2015-07-13-test/gh-pages/index.html 
(Caused by 
NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection 
object at 0x108343f28>: Failed to establish a new connection: [Errno 8] 
nodename nor servname provided, or not known',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/unittest/mock.py", 
line 1136, in patched
     return func(*args, **keywargs)
   File "/Users/gvwilson/s/amy/workshops/test/test_util.py", line 409, 
in test_wrong_url
     parse_tags_from_event_index(url)
   File "/Users/gvwilson/s/amy/workshops/util.py", line 408, in 
parse_tags_from_event_index
     response = requests.get(url)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/api.py", 
line 69, in get
     return request('get', url, params=params, **kwargs)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/api.py", 
line 50, in request
     response = session.request(method=method, url=url, **kwargs)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/sessions.py", 
line 468, in request
     resp = self.send(prep, **send_kwargs)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/sessions.py", 
line 576, in send
     r = adapter.send(request, **kwargs)
   File 
"/Users/gvwilson/anaconda/envs/py3k/lib/python3.4/site-packages/requests/adapters.py", 
line 423, in send
     raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: 
HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max 
retries exceeded with url: /test/2015-07-13-test/gh-pages/index.html 
(Caused by 
NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection 
object at 0x108343f28>: Failed to establish a new connection: [Errno 8] 
nodename nor servname provided, or not known',))

----------------------------------------------------------------------
Ran 153 tests in 23.379s

FAILED (errors=1)
Destroying test database for alias 'default'...
make: *** [test] Error 1
~~~

**Comment by pbanaszkiewicz** [comment]
@gvwilson interesting. Right now I'm working on #465 (and related) and I got rid of that test completely (not knowing that it was hitting the network). I think we can close this issue…?

**Comment by gvwilson** [comment]
Sure, as soon as tests on `develop` don't hit the network :-)

**Comment by pbanaszkiewicz** [comment]
Fixed in #465.
